### DISCUSSION TITLE
Export ErrorResponse utility class

### DISCUSSION THREAD

**Discussion by tlindsay42** [remix-run/react-router#10665] [Proposals]
I'd like for the [`ErrorResponse` private utility class](https://github.com/remix-run/react-router/blob/react-router-dom%406.14.1/packages/router/utils.ts#L1468-L1495) to be exported (ie: non-private) from `react-router-dom` to simplify error handling. I'm currently using a shared component for both route errors and deferred loader async errors (ie: `useAsyncError` values) with a switch for selecting child components based on response status, and this gap makes the implementation more complex than necessary.

**Comment by ghost** [comment]
This class should be exported.

In the official docs, there is a how-to [on setting up your error element with multiple if blocks over ErrorResponses.](https://reactrouter.com/en/main/route/error-element#throwing-responses)

And if you have it set up this way, you should also be able to handle responses thrown from effects etc. Because that is now how you propagate authentication errors etc.
